### PR TITLE
Revert Dynamic Changes

### PR DESCRIPTION
**Pull request by DorainGrey**
## What this does
This reverts the dynamic changes introduced by @d3athrow on the 28th of October and puts them back to normal previous state of the past few years.

## Why it's good
It seems like nearly every round is extended these days, the low threat generation means even relatively high pop for the servers produces low threat levels. This results in rounds being extended or a single/double traitor almost without exception. 
Midrounds are even worse affected, with it almost always being a Grue/Catbeast/Demon.

To substantiate my claims, see the following data about round types: (Counted by hand from status channel)

Revolution/Rev Squad: 
Last Rolled: 12/12/2022
Total Rolls Since Change: 4
Preceding 65 days: 11

Nuclear Operatives:
Last Rolled: 12/13/2022
Total Rolls Since Change: 8
Preceding 65 days: 26

Vampire:
Last Rolled: 12/21/2022
Total Rolls Since Change: 9
Preceding 65 days; 38

Wizard
Last Rolled: 12/18/2022
Total Rolls Since Change: 23
Preceding 65 days; 58

Space Ninja:
Last Rolled: 12/20/2022
Total Rolls Since Change: 28
Preceding 65 days;  37

Traitor: (These numbers are a rough approximation, includes admin spawns. Too many to count by hand)
Last Rolled: Today
Total Rolls Since Change: 284
Preceding 65 days;  400~

While some of this could be explained away by pop differences, I don't think the server has declined since then, in fact the opposite but I have no hard data to substantiate this, it's certainly not enough to skew the results to this degree.
I've also noticed we have a lot more head/sec play as of late which should stack the results towards higher threat and more antags.

As for the original reasoning for the changes, I believe they were made due to complaints about there being too many highlander style antags and not enough extended or low threat rounds.
Personally I'd disagree with that sentiment and I think one or two high pop rounds were cherrypicked to try make a problem bigger than it actually is, though ultimately it's all up to opinion on what level of antags is good.

However even with the goal of reducing antags in mind, I think the approach that was taken was an incorrect one.
Instead of reducing overall threat, it's far better to increase the threat/spawning conditions required for these gamemodes, for example this was done with blob here in #33381 & #33587 
(Side note the combination of these changes with the dynamic changes means that blob has not rolled since then and is effectively stealth removed)

By multiplying the curve by an arbitrary value, you invalidate most of the point of using a distribution. Additionally the current threat levels were balanced based on a threat/population ratio that is no longer the case.

Another approach would be to add a dedicated extended mode to dynamic, if this is desired and I'd happily even code it, if that's the consensus because I believe these changes have been an extreme net negative to the server and lead to less interesting gameplay overall. We have gone from rounds were anything can happen, to nearly nothing ever happens.

TLDR: Dramatically less antags since this change, see stats above. This leads to boring rounds and should be reverted/modified. There are better ways to address too many antags.

## Changelog
:cl:
 * tweak: Dynamic has been restored to a normal lorentz distribution

### CODE REVIEW

**Comment by wotisjanitor** [comment]
imageI'm not gonna read all that someone sum it up in one sentence

**Comment by dinnerwoerror** [comment]
1. i like a good r word

> imageI'm not gonna read all that someone sum it up in one sentence

2. basically about two months + a week ago there was a change that like, reduced the possible threat that could be generated, except it was so severe that it basically reduced the antagonist spawn rate to like 1/3 of what it was then

**Comment by gurfan** [comment]
circle

**Comment by xeroxemnas** [comment]
holy based.
Make spess dangerous again

**Comment by Kurfursten** [comment]
> Side note the combination of these changes with the dynamic changes means that blob has not rolled since then and is effectively stealth removed

Blob Tuesday... has failed us

**Comment by pedr69** [comment]
PARTY BLOBBERS IN THE HOUSE

**Comment by jwhitak** [comment]
Prepare for some stream of consciousness kibitzing and inane rambling. These changes will have the following effects:
# Roundstart

- tl;dr slightly more dangerous
- 17% or so higher threat level
- This means slightly higher chances of the more uncommon game modes such as blood cult/revs/wiznerds firing.
- More tators can come in on the same ruleset roll at higher threats, so ever so slightly more tators too.
- Higher threat also means higher chances at HIGHLANDER sets or Multiantag sets, but this is like 17% more likely now so still not too much difference here.
- Some gamemodes won't be too affected at all because of the "low" amount of roundstart readied up players forcing a massive threat requirement that this small increase really won't help with (only 19 players readied up? need that sweet sweet 80 threat for blob...)
- to counter: directly adjust pop requirements and ``requirement`` variable for desired/undesired rulesets
- to make insane: count all connected clients towards the ``requirement`` variable check (not to be confused with threat cost or ``required_pop``), it will make minimum threat for more dangerous antags actually go to just what it normally costs instead of being even higher because of low readied up pops

# Midrounds

- tl;dr moderately more dangerous, extremely more dangerous over time
- 66% or so higher threat level
- Directly translates to more midrounds, especially during longer rounds where dynamic fires off multiple 'cheaper' midround antags.
- We have slightly more 'cheaper' midrounds now such as pulse demon (now slotted into dynamic properly), so higher chance at a cheap option, which means more chances to roll later for more antags.
- We have a lot of people latejoining in rounds vs readied up at roundstart, so this will definitely trigger more late game antags.
- to counter: raise pop requirements on midround rulesets, most notably pulse demons and wizards
- to make insane: if no rulesets fire during roundstart (usually caused by nobody with antag turned on, resulting in 'Extended with Midrounds' which appears as Rulesets: none in the round status history), transfer some or all of the roundstart threat into midrounds. that'll show em.

**Comment by Eneocho** [comment]
> PARTY BLOBBERS IN THE HOUSE

It's "PARTY BLOB IS"

**Comment by dinnerwoerror** [comment]
> transfer some or all of the roundstart threat into midrounds. that'll show em.

you know, I think that's how it is on TG when they ported dynamic (there's only one universal threat level rolled at the start of the round vs. a roundstart threat is rolled and the separate midround threat is rolled and kept for later) and that place is fuckin powergamer central

**Comment by Kurfursten** [comment]
That's how dynamic worked originally. Midround threat being separate was a mistake.

**Comment by xeroxemnas** [comment]
more midrounds so I don't hear people screaming "MIDROUNDS WHEN" in deadchat. Also incentivizes people to latejoin more often.

**Comment by wotisjanitor** [comment]
midrounds when

**Comment by dinnerwoerror** [comment]
at the very least all the antags coming from this should give our "security" something to do other than sub as secondary antagonists

**Comment by Killette2** [comment]
Needed

**Comment by SECBATON-GRIFFON** [comment]
> PARTY BLOBBERS IN THE HOU

se tonight

**Comment by Theophrastus1** [comment]
> > PARTY BLOBBERS IN THE HOUSE
> 
> It's "PARTY BLOB IS"

image

**Comment by dinnerwoerror** [comment]
this has only been good so far, we've had our first revs and cult rounds in literal weeks and months

**Comment by Killette2** [comment]
revs and cult? I'd call that good and bad not only good
